src/textual/css/_style_properties.py

```diff
@@ -837,15 +870,18 @@ def __set__(self, obj: StylesBase, color: Color | str | None):
             if obj.set_rule(self.name, color):
                 obj.refresh()
         elif isinstance(color, str):
+            tokens = tokenize_value(color, "expression")
             try:
-                parsed_color = Color.parse(color)
+                parsed_color, is_auto = parse_color_tokens(self.name, tokens)
```

Comment:
This can raise `DeclarationError`s can't it?

This would be entirely unexpected in this context. It should raise `StyleValueError`s